test_regress/Makefile

```diff
@@ -47,11 +47,14 @@ SCENARIOS ?= --vlt --vltmt --dist
 DRIVER_HASHSET ?=
 
 .PHONY: test
-test:
+test: manifest
 	$(PYTHON3) driver.py $(DRIVER_FLAGS) $(SCENARIOS) $(DRIVER_HASHSET)
 
 ######################################################################
 
+manifest:
+	$(VERILATOR_ROOT)/bin/verilator -cc --make json -j t/t_x_assign.v && cat obj_dir/Vt_x_assign.json | jq '.version'
+
```

Comment:
This should be a test_regress test instead, e.g. t_flag_make_json.py and that test's .py file calls jq on the generated output using test.run.  Also note `jq --version` doesn't check anything as far as I know, instead you want `jq -re '""'`.

Reply:
`jq '.version'` extracts version field available in the generated JSON. I looked into that .py stuff and got lost. If you can point to some function that I can use as a template to do what you want then I'll do that.

Reply:
Try this.  Also you'll need to change the .github action script to `apt install jq`

```
import vltest_bootstrap

test.scenarios('simulator')
test.top_filename = "t/t_flag_make_cmake.v"

test.compile(verilator_flags2=['--make json'],
             verilator_make_gmake=False, verilator_make_cmake=False)

nout = test.run_capture("jq --version", check=False)
version_match = re.search(r'jq-([0-9.]+)', nout, re.IGNORECASE)
if not version_match:
    test.skip("jq is not installed")

json_filename = test.obj_dir + "/" + test.vm_prefix + ".json"
if not os.path.exists(json_filename):
    test.error(json_filename + " does not exist")

test.run(cmd=['cat "' + json_filename + '" | jq ".version"'])

test.passes()
```

Reply:
JQ is mentioned there https://github.com/verilator/verilator/blob/master/docs/guide/install.rst?plain=1#L149 so I don't think that I need to explicitly mention it one time more. But if you have something specic in your mind then let me know.

Reply:
BTW, thanks for the .py - it would take me too much to get to that!